packages/eslint-plugin-sui/src/rules/commonjs.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * @fileoverview Ensure your code is not using CommonJS signatures like module.exports or moduel.exports.foo or require() or require.resolve()
+ */
+'use strict'
+
+const dedent = require('string-dedent')
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+/** @type {import('eslint').Rule.RuleModule} */
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Ensure that your code is using ems over commonjs modules',
+      recommended: true,
+      url: 'https://github.mpi-internal.com/scmspain/es-td-agreements/blob/master/30-Frontend/00-agreements'
+    },
+    fixable: null,
+    schema: [],
+    messages: {
+      forbiddenExports: dedent`
+        Use module.* should be avoid.
+      `,
+      forbiddenRequires: dedent`
+        Use require function should be avoid.
```

Comment:
What if anyone is using a `require` in a ESM compliant file? e.g.:
```
import { createRequire } from "module"
const require = createRequire(import.meta.url)

require('whatever')
```

Should we allow it or not?

Reply:
Doc reference: https://nodejs.org/api/module.html#modulecreaterequirefilename